spark/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala

```diff
@@ -1434,6 +1434,15 @@ trait DeltaSQLConfBase {
       .booleanConf
       .createWithDefault(true)
 
+  val MDC_SORT_WITHIN_PARTITIONS =
+    SQLConf.buildConf("spark.databricks.io.skipping.mdc.sortWithinPartitions")
+      .internal()
+      .doc("If enabled, partitions are sorted on Z-order values for MDC. " +
+         "This co-locates records with the same Z-order values in row groups, " +
+         "which enables data skipping on the Parquet level.")
```

Comment:
```suggestion
    SQLConf.buildConf("spark.databricks.io.skipping.mdc.sortWithinFiles")
      .internal()
      .doc("If enabled, sort within files by the specified MDC curve. " +
         "This might improve row-group skipping and data compression, at " +
         "the cost of additional overhead for sorting.")
```
two nits:
1. can we rename this to sortWithinFiles, since partitions are so overloaded (in this case it means files, but can also mean hive partition, etc).
2. MDC also includes Hilbert curve, so best to not mention Z-order here.